### ISSUE TITLE
Formal spec on binaryen text input?

### ISSUE THREAD

**Issue by bobzhang** [WebAssembly/binaryen#5591]
I was aware of [this](https://github.com/WebAssembly/binaryen/wiki/Compiling-to-WebAssembly-with-Binaryen#what-do-i-need-to-have-in-order-to-use-binaryen-to-compile-to-webassembly) but do we have a formal spec of the text syntax? 
Or using the binaryen API is the *only* recommended way to use binaryen? Thank you

**Comment by tlively** [comment]
No, unfortunately we don't have a comprehensive list of the differences between Binaryen's text format and the [standard WebAssembly text format](https://webassembly.github.io/spec/core/text/index.html). I have a side project to replace Binaryen's text parser so that it will use the standard text format, but that's not finished.

In the meantime, you can look at Binaryen's tests to see what kind of text format it expects (although some things might need to be tweaked once we switch to parsing the standard text format).

Besides using the C API, the most stable way to use Binaryen will always be to pass it a WebAssembly binary, since we already support the standard binary format and it has strict backward compatibility guarantees.

**Comment by tlively** [comment]
Binaryen supports the standard text format now.
